src/Plant/dynamic/update_tree_with_time.jl

```diff
@@ -0,0 +1,75 @@
+using Parameters
+
+# function to update the tree with time, default at a Δt=1s using Wang-2020 scheme
+function update_tree_with_time(tree::StructTree, Δt::FT=FT(1.0); scheme::String="Wang2020", updating=false) where {FT}
```

Comment:
Can we add doc strings to this? [here](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Common/MoistThermodynamics/relations.jl#L73) is a good example of a doc string that follows our coding style